### DISCUSSION TITLE
Need help with defining onSuccess callback for useSWR

### DISCUSSION THREAD

**Discussion by sfdcale** [vercel/next.js#12414] [Help] [answered]
I posted this question here(https://github.com/zeit/swr/issues/362) 2 days ago but didn't receive any response, hence posting here. 

I am trying to use `onSuccess` callback function on useSWR with no success, here is the simplified but reproducible code:

```
useSWR("/api/data", fetcher, { onSuccess: 
    (data, key, config) => {
      console.log('Entered onSuccess'); 
      console.log({data}); //This always prints undefined.
    }
  }
);
```

**/api/data.js:**
```
export default (req, res) => {
    res.statusCode = 200;
    res.setHeader('Content-Type', 'application/json');
    res.send({ FirstName: "John", LastName: "Doe"});
}
```
**package.json**:
```
"next": "^9.0.6",
"swr": "^0.2.0"
```

Here is the codesandbox showing the problem : https://codesandbox.io/s/nervous-hill-mn0kz?file=/pages/index.js

**Comment by timneutkens** [comment, marked as answer]
Was answered here: https://github.com/zeit/swr/issues/362
